src/OrbitCaptureClient/include/OrbitCaptureClient/GpuQueueSubmissionProcessor.h

```diff
@@ -85,13 +85,15 @@ class GpuQueueSubmissionProcessor {
   [[nodiscard]] static std::optional<orbit_grpc_protos::GpuCommandBuffer> ExtractFirstCommandBuffer(
       const orbit_grpc_protos::GpuQueueSubmission& gpu_queue_submission);
 
-  [[nodiscard]] const orbit_grpc_protos::GpuJob* FindMatchingGpuJob(
+  // Finds the GpuJob that is fully inside the given timestamps and happened on the given thread id.
+  // Returns `nullopt` if there is no such job.
+  [[nodiscard]] std::optional<orbit_grpc_protos::GpuJob> FindMatchingGpuJob(
       int32_t thread_id, uint64_t pre_submission_cpu_timestamp,
       uint64_t post_submission_cpu_timestamp);
 
   // Finds the GpuQueueSubmission that fully contains the given timestamp and happened on the given
-  // thread id. Returns `nullptr` if there is no such submission.
-  [[nodiscard]] const orbit_grpc_protos::GpuQueueSubmission* FindMatchingGpuQueueSubmission(
+  // thread id. Returns `nullopt` if there is no such submission.
+  [[nodiscard]] std::optional<orbit_grpc_protos::GpuQueueSubmission> FindMatchingGpuQueueSubmission(
       int32_t thread_id, uint64_t submit_time);
   [[nodiscard]] bool HasUnprocessedBeginMarkers(int32_t thread_id,
                                                 uint64_t post_submission_timestamp) const;
```

Comment:
out of curiosity does changing flat_hash_map to node_hash_map help?

Reply:
Well, the issue is in the inner std::map (which I use for the ordering (upper_bound/lower_bound).

Reply:
still if this is an implicit move by outer map it could do something to the outer container. It could be some implementation specific internal data representation of inner map which is tied to an instance and might get cleared if outer container is modified.

Reply:
I would still use node_map, it is pointer to a map which has pointer stability but I still think moving map does not have to provide same guarantee. It likely does but still.

Reply:
Done.